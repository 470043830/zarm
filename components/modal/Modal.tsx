import React, { Component } from 'react';
import classnames from 'classnames';
import { BaseModalProps } from './PropsType';
import Popup from '../popup';

export interface ModalProps extends BaseModalProps {
  prefixCls?: string;
  className?: string;
}

export default class Modal extends Component<ModalProps, any> {
  static Header: any;

  static Body: any;

  static Footer: any;

  static defaultProps = {
    prefixCls: 'za-modal',
    visible: false,
    animationType: 'fade',
    animationDuration: 200,
    width: '70%',
    shape: 'radius',
  };

  render() {
    const { prefixCls, className, shape, children, ...others } = this.props;

    const cls = {
      modal: classnames(prefixCls, className, {
        [`${prefixCls}--${shape}`]: !!shape,
      }),
    };

    return (
      <Popup
        className={cls.modal}
        direction="center"
        {...others}
      >
        {children}
      </Popup>
    );
  }
}
