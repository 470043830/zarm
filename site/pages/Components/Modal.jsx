import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/modal/demo.md'),
      className: 'modal-page',
    };
  }
}
