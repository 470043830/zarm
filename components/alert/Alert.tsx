import React, { PureComponent } from 'react';
import ReactDOM from 'react-dom';
import classnames from 'classnames';
import PropsType from './PropsType';
import Modal from '../modal';
import alertLocale from './locale/zh_CN';
import { getRunTimeLocale } from '../locale-provider/LocaleProvider';

export interface AlertProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Alert extends PureComponent<AlertProps, {}> {
  static defaultProps = {
    prefixCls: 'za-alert',
    animationType: 'zoom',
    message: '',
    locale: alertLocale,
  };

  // private static alertContainer;

  // static show = (props: AlertProps) => {
  //   const { defaultProps } = Alert;
  //   let _props: any;

  //   if (typeof props === 'object') {
  //     _props = { ...defaultProps, ...props };
  //   } else {
  //     _props = { ...defaultProps };
  //   }

  //   const runTimeLocale = getRunTimeLocale();
  //   if (runTimeLocale && runTimeLocale.Alert) {
  //     _props = { ..._props, locale: runTimeLocale.Alert };
  //   }
  //   Alert.alertContainer = document.createElement('div');
  //   document.body.appendChild(Alert.alertContainer);
  //   return new Promise((resolve) => {
  //     ReactDOM.render(
  //       <Alert
  //         {..._props}
  //         visible
  //         onCancel={() => {
  //           if (typeof _props.onCancel === 'function') {
  //             _props.onCancel();
  //           }
  //           resolve(false);
  //         }}
  //       />,
  //       Alert.alertContainer,
  //     );
  //   });
  // };

  // static hide = () => {
  //   if (Alert._hide) {
  //     Alert._hide();
  //   }
  // };

  // static _hide: () => void;

  state = {
    visible: this.props.visible,
  };

  // componentDidMount() {
  //   Alert._hide = this._hide;
  // }

  componentWillReceiveProps(nextProps) {
    const { visible } = this.props;

    if (nextProps.visible !== visible) {
      if (nextProps.visible === true) {
        this.setState({
          visible: true,
        });
      } else {
        this._hide();
      }
    }
  }

  _hide = () => {
    this.setState({
      visible: false,
    });
  };

  afterClose = () => {
    const { afterClose } = this.props;
    // if (Alert.alertContainer) {
    //   document.body.removeChild(Alert.alertContainer);
    //   Alert.alertContainer = null;
    // }

    if (typeof afterClose === 'function') {
      afterClose();
    }
  };

  render() {
    const { prefixCls, className, message, cancelText, onCancel, locale, ...others } = this.props;
    const { visible } = this.state;
    const cls = classnames(prefixCls, className);

    return (
      <Modal
        className={cls}
        {...others}
        visible={visible}
        afterClose={this.afterClose}
        footer={<div className={`${prefixCls}__button`} onClick={onCancel}>{cancelText || locale!.cancelText}</div>}
      >
        {message}
      </Modal>
    );
  }
}
