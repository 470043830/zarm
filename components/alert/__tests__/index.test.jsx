import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Alert from '../index';
import Modal from '../../modal';

describe('Alert', () => {
  it('renders correctly', () => {
    const onCancel = jest.fn();

    const wrapper = render(
      <Alert
        radius
        visible
        title="警告"
        message="这里是警告信息"
        onCancel={onCancel}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with static methods', () => {
    const onCancel = jest.fn();
    const modal = Modal.alert({
      title: '静态调用的title',
      message: '静态调用的body，使用promise关闭',
      onCancel,
    });
    // const promise = Alert.show({
    //   title: '警告',
    //   message: '这是警告信息',
    //   onCancel,
    // });
    document.querySelector('.za-alert__button').click();

    expect(onCancel).toBeCalledTimes(1);
    expect(modal).resolves.toBe(true);
  });

  // it('close with static methods', () => {
  //   const modal = Modal.alert({
  //     title: '静态调用的title',
  //     message: '静态调用的body，使用promise关闭',
  //   });

  //   document.querySelector('.za-alert__button').click();
  //   expect(modal).resolves.toBe(true);
  // });

  // it('close with async methods', () => {
  //   const onCancel = jest.fn(() => Promise.resolve(true));
  //   const modal = Modal.alert({
  //     title: '静态调用的title',
  //     message: '静态调用的body，使用promise关闭',
  //     onCancel,
  //   });

  //   onCancel();
  //   expect(modal).resolves.toBe(true);
  // });
});
