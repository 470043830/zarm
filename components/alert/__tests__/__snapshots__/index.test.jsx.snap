// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  class="za-modal za-alert za-modal--radius"
  style="-webkit-animation-duration:200ms;animation-duration:200ms"
>
  <div
    class="za-modal__wrapper"
  >
    <div
      class="za-modal__dialog"
      style="width:70%;-webkit-animation-duration:200ms;animation-duration:200ms"
    >
      <div
        class="za-modal__header"
      >
        <div
          class="za-modal__header__title"
        >
          警告
        </div>
      </div>
      <div
        class="za-modal__body za-alert__body"
      >
        这里是警告信息
      </div>
      <div
        class="za-modal__footer za-alert__footer"
      >
        <a
          class="za-alert__footer__button"
        >
          Close
        </a>
      </div>
    </div>
  </div>
  <div
    class="za-mask za-mask--normal"
  />
</div>
`;
